ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -239,4 +239,94 @@ public static synchronized String getSchemaDefinitionString(Schema schema) {
     return stringBuilder.toString();
   }
 
+  public static String getSQLTypeName(Schema schema) {
+    switch (schema.type()) {
+      case INT32:
+        return "INT";
+      case INT64:
+        return "BIGINT";
+      case FLOAT32:
+      case FLOAT64:
+        return "DOUBLE";
+      case BOOLEAN:
+        return "BOOLEAN";
+      case STRING:
+        return "VARCHAR";
+      case ARRAY:
+        return "ARRAY<" + getSQLTypeName(schema.valueSchema()) + ">";
+      case MAP:
+        return "MAP<" + getSQLTypeName(schema.keySchema()) + "," + getSQLTypeName(schema
+                                                                                       .valueSchema()) + ">";
+      default:
+        throw new KsqlException(String.format("Invalid type in schema: %s.", schema.toString()));
+    }
+  }
+
+  public static String buildAvroSchema(final Schema schema, String name) {
+
+    org.apache.avro.SchemaBuilder.FieldAssembler fieldAssembler = org.apache.avro.SchemaBuilder
+        .record(name).namespace("ksql")
+        .fields();
+    for (Field field : schema.fields()) {
+      fieldAssembler.name(field.name().replace(".", "_")).type(getAvroSchema(field.schema())).noDefault();
+    }
+
+    return fieldAssembler.endRecord().toString();
+  }
+
+  private static org.apache.avro.Schema getAvroSchema(Schema schema) {
```

Comment:
I think a clear name would be `getAvroSchemaForField`, and name the argument `fieldSchema`. Otherwise it looks at the outset that you are translating an entire connect schema to an avro schema.

Reply:
Sounds good, changed it.